Packs/OktaAuth0/Integrations/OktaAuth0EventCollector/OktaAuth0EventCollector.py

```diff
@@ -0,0 +1,292 @@
+
+
+''' HELPER FUNCTIONS '''
+
+
+def arg_to_strtime(value: Any) -> Optional[str]:
+    if datetime_obj := arg_to_datetime(value):
+        return datetime_obj.strftime(DATE_FORMAT)
+    return None
+
+
+def prepare_query_params(params: dict, last_run: dict = {}) -> dict:
+    """
+    Parses the given inputs into Okta Auth0 Events API expected params format.
+    """
+    query_params = {}
+
+    if not last_run:  # requesting by time query
+        since = arg_to_strtime(params.get('since'))
+        query_params = {
+            "q": last_run.get('query') or f"date:[{since} TO *]",
+            "sort": "date:1",
+            "per_page": 100
+        }
+
+    else:  # requesting by log_id
+        query_params = {
+            "from": last_run.get('last_id'),
+            "sort": "date:1",
+            "take": 100
+        }
+
+    return query_params
+
+
+def should_refresh_access_token() -> bool:
+    """Check whether the access token in the integration context has expired.
+
+    Returns:
+        bool: Whether the token has expired and should refresh the token.
+    """
+    int_context = get_integration_context()
+    if 'expired_token_time' not in int_context or \
+            int(time.time()) - int_context.get('expired_token_time', 1) > EXPIRED_TOKEN_RANGE:
+        return True
+    return False
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def __init__(self, base_url, client_id, client_secret, verify, proxy):
+        super().__init__(base_url, verify, proxy, headers={})
+        self.client_id = client_id
+        self.client_secret = client_secret
+
+    def get_access_token_request(self):
+        """
+        Send request to get an access token from Okta Auth0.
+        """
+        body = {
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+            "audience": f"{self._base_url}/api/v2/",
+            "grant_type": "client_credentials"
+        }
+
+        raw_response = self._http_request(
+            method='POST',
+            url_suffix='/oauth/token',
+            data=body
+        )
+
+        demisto.info('Successfully got access token.')
+        return raw_response
+
+    def get_access_token(self):
+        """Get the access token from the integration context if it has not expired.
+
+        Returns:
+            str: The access token.
+        """
+        if should_refresh_access_token():
+            raw_response = self.get_access_token_request()
+            set_integration_context({
+                "access_token": raw_response.get("access_token"),
+                "expired_token_time": int(time.time())
+            })
+
+        return get_integration_context()["access_token"]
+
+    def get_events_request(self, query_params: dict) -> list:
+        """
+        Send request to get events from Okta Auth0.
+        """
+        access_token = self.get_access_token()
+        self._headers.update({'Authorization': f'Bearer {access_token}'})
```

Comment:
let's put these in the c'tor instead (in case we expand the collector with other commands in the future, e.g. fetch-assets)

Reply:
done